### PR TITLE
Issue 15 add proto generation

### PR DESCRIPTION
**Pull request by mhirna**
This pull request implements the following:

Adding collection nodes.
Adding proto generation:
- config.proto
- autogenerated python script
- proto_graph.py, that generates proto graph.

### CODE REVIEW

File: graph_generation/proto_graph.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        Saves generated graph message to binary. If overwrite - existing file will be overwritten.
+
+        Raises:
+            ValueError: Graph database with this file already exists.
+        """
+        if os.path.isfile(filename) and overwrite:
+            os.remove(filename)
+        elif os.path.isfile(filename):
+            raise ValueError("Graph database with this file already exists.")
+        with open(filename, "ab") as f:
```
Comment (ZikiSoy) [inline graph_generation/proto_graph.py]: Should use "wb" mode instead?

File: graph_generation/proto_graph.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            "NOT_CRITICAL": config_pb2.ProtoGraph.System.SystemCriticality.NOT_CRITICAL,
+            "CRITICAL_CAN_CAUSE_S0_OUTAGE": config_pb2.ProtoGraph.System.SystemCriticality.CRITICAL_CAN_CAUSE_S0_OUTAGE,
+            "CRITICAL_SIGNIFICANT_RUN_RATE": config_pb2.ProtoGraph.System.SystemCriticality.CRITICAL_SIGNIFICANT_RUN_RATE,
+            "CRITICAL_OTHER": config_pb2.ProtoGraph.System.SystemCriticality.CRITICAL_OTHER
+        }
+        if system_criticality in system_criticality_dict:
+            return system_criticality_dict[system_criticality]
+        else:
+            logging.error("Incorrect system criticality value. Setting NEVER as a default.")
+            return system_criticality_dict["NEVER"]
```
Comment (ZikiSoy) [inline graph_generation/proto_graph.py]: "NEVER" is not defined in the dict.

**Review by ZikiSoy — COMMENTED** [review]
Maybe consider adding __pycache__ files to gitignore
https://github.com/github/gitignore/blob/master/Python.gitignore

File: graph_generation/proto_graph.py (hunk 1 above)
Comment (mhirna) [reply graph_generation/proto_graph.py]: Yes, wb is the correct way, will fix

**Review by ZikiSoy — APPROVED** [review]

**Pull request merged**